src/onegov/org/cronjobs.py

```diff
@@ -459,3 +462,50 @@ def form(definition):
             receivers=(address, ),
             content=content
         )
+
+
+@OrgApp.cronjob(hour=4, minute=30, timezone='Europe/Zurich')
+def archive_old_tickets(request):
+
+    archive_timespan = request.app.org.auto_archive_timespan
+
+    session = request.session
+
+    if archive_timespan is None:
+        return
+
+    if archive_timespan == 0:
+        return
+
+    archive_timespan = timedelta(days=archive_timespan)
+
+    diff = utcnow() - archive_timespan
+    query = session.query(Ticket)
+    query = query.filter(Ticket.state == 'closed')
+    query = query.filter(Ticket.created <= diff)
+
+    for ticket in query:
+        if getattr(ticket.handler, 'has_future_reservation', False):
```

Comment:
This is not correct, a method is always truthy, you actually need to call the method:
```suggestion
        if getattr(ticket.handler, 'has_future_reservation', lambda: False)():
```

Reply:
If you decide to change this to a `cached_property`, like suggested further below, then this doesn't need to change.

Reply:
Good catch here. If we change `has_future_reservation` to be cached_property, this ` lambda: False)()`  would not be necessary ?

Reply:
correct

Reply:
That would also be a good case for a linter. I'm actually kind of surprised

Reply:
mypy would catch this, but as long as this doesn't have type annotations it will not.

Reply:
Actually nevermind the `getattr` just inherently makes this difficult to statically analyze.

Reply:
I was contemplating using `isinstance` anyway. Do you know if there's not much of a performance difference between that and `getattr` ?

Reply:
shouldn't be a big hit, there's some things like runtime checkable protocols which would be slow in an `isinstance` check and there are other classes that implement their own `__instancecheck__` which has a potential to be slow, but this shouldn't apply here.